### ISSUE TITLE
v0.0.1 proof-of-concept

### ISSUE THREAD

**Issue by armanbilge** [armanbilge/huckle#1]
As a proof-of-concept, my goal is to develop a `huckle cli`, as a sort of scala-cli clone. The goal is not to clone scala-cli, just to prototype some ideas :)

- it will be published to npm, so you can do stuff like this, on virtually any machine:
    - `npx huckle cli run hello.scala`
    - `npm i -g huckle`
    - `huckle cli compile hello.scala`
- it will have a maven artifact resolver for JVM/JS/Native
- it will have an incremental compiler
    - `scalac` is JVM only, so it will be invoked by a `huckle-server` JVM application published to maven central
    - the server will be a long-running daemon shared across invocations and builds
    - smithy4s will define the interface, and the protocol can be over unix sockets
    - ideally the server will be stateless, and just a thin wrapper over dotty compiler API
    - in which case, the client will be responsible for coordinating incremental compilation
    - I'd like to explore using SQLite for persisting incremental compile data
- it will support `compile` and `run`, and the `using lib` and `using option` directives

In summary, the `huckle cli` will be a Node.js script that:
- fetches the `huckle-server` artifact from maven, if necessary
- starts the `huckle-server` daemon with `$JAVA_HOME`, if necessary
- parses the directives, and resolves any dependencies
- incrementally (re-)compiles the target source using the daemon server
- runs the compiled program
